src/h3/mod.rs

```diff
@@ -760,7 +755,13 @@ impl Connection {
             return Err(Error::Done);
         }
 
-        Ok(stream.try_consume_data(conn, out)?)
+        let read = stream.try_consume_data(conn, out)?;
+
+        if conn.stream_finished(stream_id) {
+            self.finished_streams.push_back(stream_id);
```

Comment:
My spidersense was tingling at this code being here. I had to trace back through the code to remember that DATA frame case wouldn't be handled by other functions and that it is required here. Maybe worth a comment?

Reply:
I'm not sure I understand what clarification is needed here, any suggestion for what the comment should say?

Reply:
I was thinking of something like 

```
While the in stream::State::Data, stream closure needs to be detected and handled when body bytes are read
```

but I don't know how much that helps anyone coming to this code.

Reply:
Ok, added a comment now.